examples/crm/src/dataProvider.ts

```diff
@@ -82,11 +83,99 @@ const dataProviderWithCustomMethod = {
     },
+    transferAdministrator: async (from: Identifier, to: Identifier) => {
+        const { data: sales } = await baseDataProvider.getList('sales', {
+            filter: { id: [from, to] },
+            pagination: { page: 1, perPage: 2 },
+            sort: { field: 'name', order: 'ASC' },
+        });
+
+        const fromSale = sales.find(sale => sale.id === from);
+        const toSale = sales.find(sale => sale.id === to);
+
+        if (!fromSale || !toSale) {
+            return;
+        }
+
+        await baseDataProvider.update('sales', {
+            id: to,
+            data: {
+                administrator: true,
+            },
+            previousData: toSale,
+        });
+
+        return await baseDataProvider.update('sales', {
+            id: from,
+            data: {
+                administrator: false,
+            },
+            previousData: fromSale,
+        });
+    },
 };
 
+export type CustomDataProvider = typeof dataProviderWithCustomMethod;
+
 export const dataProvider = withLifecycleCallbacks(
     dataProviderWithCustomMethod,
     [
+        {
+            resource: 'sales',
+            afterDelete: async (params, dataProvider) => {
+                const [companies, contacts, contactNotes, deals] =
+                    await Promise.all([
+                        dataProvider.getList('companies', {
+                            filter: { sales_id: params.data.id },
+                            pagination: {
+                                page: 1,
+                                perPage: 10_000,
+                            },
+                            sort: { field: 'id', order: 'ASC' },
+                        }),
+                        dataProvider.getList('contacts', {
+                            filter: { sales_id: params.data.id },
+                            pagination: {
+                                page: 1,
+                                perPage: 10_000,
+                            },
+                            sort: { field: 'id', order: 'ASC' },
+                        }),
+                        dataProvider.getList('contactNotes', {
+                            filter: { sales_id: params.data.id },
+                            pagination: {
+                                page: 1,
+                                perPage: 10_000,
+                            },
+                            sort: { field: 'id', order: 'ASC' },
+                        }),
+                        dataProvider.getList('deals', {
+                            filter: { sales_id: params.data.id },
+                            pagination: {
+                                page: 1,
+                                perPage: 10_000,
+                            },
+                            sort: { field: 'id', order: 'ASC' },
+                        }),
+                    ]);
+
+                await Promise.all([
+                    dataProvider.deleteMany('companies', {
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** 
We don't want to delete the linked resources, but update them so that they are linked to the administrator.
‏